exploration/number-selection.md

```diff
@@ -75,6 +90,64 @@ either plural or ordinal selection in a single message.
 > * {{You have {$numRemaining} chances remaining (plural)}}
 >```
 
+As a user, I want the selector to match the options specified:
+```
+.local $num = {123.456 :number maximumSignificantDigits=2 maximumFractionDigits=2 minimumFractionDigits=2}
+.match {$num}
+120.00  {{This matches}}
```

Comment:
The exact matching is rarely used, and not for situations like this.

It is usually handy for cases like "1 {this is the last day}" or "1 {you got the gold medal}"
In these cases the value itself is not even rendered in the message.

Cases like "1 dollar" / "1.00 dollars" are handled properly by the plural rules and they don't need exact matches (`one {{$amount} dollar" other "{$amount}  dollars}`)

Reply:
You might be thinking too narrowly about usage. Yes, a lot of custom messages are of the "You have no..." (aka `=0`) or "This is your last..." (aka `=1`) variety. But there are also many messages that _do_ insert the value (rather than forcing the translator to type it, e.g. "You received a reward when you accumulated {$points} points.") or cases where the message is tied to other (still-specific) values. "Many" is a relative term, of course. There are orders of magnitude fewer of these types of messages than plain `0`/`1` messages and orders of magnitude fewer of those than "insert number here" plural category ones. That doesn't make them unimportant.

In fact, to me that makes them _more_ important, since those messages should work consistently with user experience of simpler values. People make mistakes when they can't just do what they would do "normally".